### PR TITLE
Allow type inference to check for types in superclasses.

### PR DESCRIPTION
**Pull request by Ambrevar**
@aartaka OK to merge?

### CODE REVIEW

**Comment by Ambrevar** [comment]
Note that this drags `moptilities` into the deps of the main system.

File: source/defclass-star.lisp [hunk 1]
```diff
@@ -64,6 +64,10 @@ subtypes, e.g.  for \"\" return 'string instead of `(SIMPLE-ARRAY CHARACTER
 Note that in a slot definition, '() is inferred to be a list while NIL is
 inferred to be a boolean.
 
+If the slot is found in a finalized superclass, the inferred type is then that
+of the class.  If some superclass is not finalized, no type inference is
```
Comment (aartaka) [inline source/defclass-star.lisp:68]: "of the superclass"?

File: source/defclass-star.lisp [hunk 2]
```diff
@@ -455,7 +480,7 @@ New class options (defaults are NIL unless specified):
 - `:type-inference' (default: `default-type-inference').
 
   Function that returns the type of a slot definition (with an initform).  It
-takes 3 arguments:
+  takes 3 arguments:
```
Comment (aartaka) [inline source/defclass-star.lisp:483]: Some indentation is off?

**Review by aartaka — COMMENTED** [review]
All good, except for some minor documentation thing :)

File: source/defclass-star.lisp (hunk 1 above)
Comment (Ambrevar) [reply source/defclass-star.lisp:68]: Where exactly?

File: source/defclass-star.lisp (hunk 2 above)
Comment (Ambrevar) [reply source/defclass-star.lisp:483]: It was wrong before I think.

Comment (aartaka) [reply source/defclass-star.lisp:483]: Ah, I see now. The format is not exactly cool (I'd personally do something like Shinmera-styled formatting), but we can live with it :)

File: source/defclass-star.lisp (hunk 1 above)
Comment (aartaka) [reply source/defclass-star.lisp:68]: "If the slot is found in a finalized superclass, the inferred type is then that of the _superclass_."

Comment (Ambrevar) [reply source/defclass-star.lisp:68]: Oh right!

**Comment by aartaka** [comment]
It's mergeable now (★^O^★)

**Pull request merged**